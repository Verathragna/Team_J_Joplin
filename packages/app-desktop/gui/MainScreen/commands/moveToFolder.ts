import { CommandRuntime, CommandDeclaration, CommandContext } from '@joplin/lib/services/CommandService';
import { _ } from '@joplin/lib/locale';
import Folder from '@joplin/lib/models/Folder';
import Note from '@joplin/lib/models/Note';

export const declaration: CommandDeclaration = {
	name: 'moveToFolder',
	label: () => _('Move to notebook'),
};

export class AddOptions {

	private maxDepth = 15;
	public startFolders: any[] = [];

	public addOptions(folders: any[], depth: number, path: string) {
		for (let i = 0; i < folders.length; i++) {
			const folder = folders[i];
			const newPath = path + folder.title;
			this.startFolders.push({ key: folder.id, value: folder.id, label: folder.title, title: folder.title, path: newPath, indentDepth: depth, _indentDepth: depth });
			if (folder.children) this.addOptions(folder.children, (depth + 1) < this.maxDepth ? depth + 1 : this.maxDepth, `${newPath}/`);
		}
	}
}

export const runtime = (comp: any): CommandRuntime => {
	return {
		execute: async (context: CommandContext, noteIds: string[] = null) => {
			noteIds = noteIds || context.state.selectedNoteIds;

			const folders: any[] = await Folder.sortFolderTree();
			const options = new AddOptions();
			options.addOptions(folders, 0, '');

			comp.setState({
				promptOptions: {
					label: _('Move to notebook:'),
					inputType: 'dropdown',
					value: '',
					autocomplete: options.startFolders,
					onClose: async (answer: any) => {
						if (answer) {
							for (let i = 0; i < noteIds.length; i++) {
								await Note.moveToFolder(noteIds[i], answer.value);
							}
						}
						comp.setState({ promptOptions: null });
					},
				},
			});
		},
		enabledCondition: 'someNotesSelected',
	};
};
