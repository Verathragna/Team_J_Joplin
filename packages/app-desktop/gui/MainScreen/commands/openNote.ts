import { CommandRuntime, CommandDeclaration, CommandContext } from '@joplin/lib/services/CommandService';
import Note from '@joplin/lib/models/Note';
import Folder from '@joplin/lib/models/Folder';
import Setting from '@joplin/lib/models/Setting';


export const declaration: CommandDeclaration = {
	name: 'openNote',
};

export const runtime = (): CommandRuntime => {
	return {
		execute: async (context: CommandContext, noteId: string, hash: string = null) => {
			const note = await Note.load(noteId);
			if (!note) throw new Error(`No such note: ${noteId}`);

			const folder = await Folder.load(note.parent_id);
			if (!folder) throw new Error(`Note parent notebook does not exist: ${JSON.stringify(note)}`);

			Setting.openDefaultEditor();

			context.dispatch({
				type: 'FOLDER_AND_NOTE_SELECT',
				folderId: folder.id,
				noteId: note.id,
				hash,
			});
		},
	};
};
