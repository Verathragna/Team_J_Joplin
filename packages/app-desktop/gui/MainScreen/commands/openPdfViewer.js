'use strict';
const __awaiter = (this && this.__awaiter) || function(thisArg, _arguments, P, generator) {
	function adopt(value) { return value instanceof P ? value : new P(function(resolve) { resolve(value); }); }
	return new (P || (P = Promise))(function(resolve, reject) {
		function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
		function rejected(value) { try { step(generator['throw'](value)); } catch (e) { reject(e); } }
		function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
		step((generator = generator.apply(thisArg, _arguments || [])).next());
	});
};
Object.defineProperty(exports, '__esModule', { value: true });
exports.runtime = exports.declaration = void 0;
const locale_1 = require('@joplin/lib/locale');
const Resource_1 = require('@joplin/lib/models/Resource');
exports.declaration = {
	name: 'openPdfViewer',
	label: () => locale_1._('Open PDF viewer'),
};
exports.runtime = () => {
	return {
		execute: (context, resourceId, pageNo) => __awaiter(void 0, void 0, void 0, function* () {
			const resource = yield Resource_1.default.load(resourceId);
			if (!resource) { throw new Error(`No such resource: ${resourceId}`); }
			if (resource.mime !== 'application/pdf') { throw new Error(`Not a PDF: ${resource.mime}`); }
			console.log('Opening PDF', resource);
			context.dispatch({
				type: 'DIALOG_OPEN',
				name: 'pdfViewer',
				props: {
					resource,
					pageNo: pageNo || 1,
				},
			});
		}),
	};
};
// # sourceMappingURL=openPdfViewer.js.map
