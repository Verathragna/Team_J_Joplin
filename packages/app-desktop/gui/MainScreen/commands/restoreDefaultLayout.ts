import { CommandRuntime, CommandDeclaration } from '@joplin/lib/services/CommandService';
import { _ } from '@joplin/lib/locale';
import dialogs from '../../dialogs';
// import Setting from '@joplin/lib/models/Setting';

export const declaration: CommandDeclaration = {
	name: 'restoreDefaultLayout',
	label: () => _('Restore default layout'),
};

export const runtime = (): CommandRuntime => {
	return {
		execute: async () => {
			const message = _('Are you sure you want to return to the default layout? The current layout configuration will be lost.');
			const confirmed = await dialogs.confirm(message);

			if (!confirmed) return;

			// load the default values
			// set as the new active ones
			// log a event??
		},
	};
};
