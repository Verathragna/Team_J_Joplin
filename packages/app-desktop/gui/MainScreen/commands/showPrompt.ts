import { CommandRuntime, CommandDeclaration, CommandContext } from '@joplin/lib/services/CommandService';

export const declaration: CommandDeclaration = {
	name: 'showPrompt',
};

interface PromptConfig {
	label: string;
	inputType?: 'dropdown' | 'datetime' | 'tags' | 'text';
	value?: any;
	autocomplete?: any[];
	buttons?: string[];
}

export const runtime = (comp: any): CommandRuntime => {
	return {
		execute: async (_context: CommandContext, config: PromptConfig) => {
			return new Promise((resolve) => {
				comp.setState({
					promptOptions: {
						...config,
						onClose: async (answer: any, buttonType: string) => {
							comp.setState({ promptOptions: null });
							resolve({
								answer: answer,
								buttonType: buttonType,
							});
						},
					},
				});
			});
		},
	};
};
