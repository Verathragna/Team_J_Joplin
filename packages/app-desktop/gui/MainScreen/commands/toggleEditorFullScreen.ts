import CommandService, {
	CommandContext,
	CommandDeclaration,
	CommandRuntime,
} from '@joplin/lib/services/CommandService';
import { _ } from '@joplin/lib/locale';
import setLayoutItemProps from '../../ResizableLayout/utils/setLayoutItemProps';
import layoutItemProp from '../../ResizableLayout/utils/layoutItemProp';
import { AppState } from '../../../app.reducer';

// Record the current state of the sidebar and note list
const lastUserLayoutStatus = {
	isNoteListVisible: true,
	isSidebarVisible: true,
};

export const declaration: CommandDeclaration = {
	name: 'toggleEditorFullScreen',
	label: () => _('Toggle fullscreen'),
	iconName: 'fas fa-expand',
};

export const runtime = (): CommandRuntime => {
	return {
		execute: async (context: CommandContext) => {
			const layout = (context.state as AppState).mainLayout;

			const isNoteListVisible = layoutItemProp(layout, 'noteList', 'visible');
			const isSidebarVisible = layoutItemProp(layout, 'sideBar', 'visible');
			let isEditorFullScreen = layoutItemProp(layout, 'editor', 'isFullScreen');

			if (isEditorFullScreen === undefined) {
				isEditorFullScreen = !isNoteListVisible && !isSidebarVisible;
			}

			let newLayout = layout;

			if (isEditorFullScreen) {
				newLayout = await CommandService.instance().execute('toggleSideBar', lastUserLayoutStatus.isSidebarVisible);
				newLayout = await CommandService.instance().execute('toggleNoteList', lastUserLayoutStatus.isNoteListVisible);
			} else {
				lastUserLayoutStatus.isSidebarVisible = isSidebarVisible;
				lastUserLayoutStatus.isNoteListVisible = isNoteListVisible;

				newLayout = await CommandService.instance().execute('toggleSideBar', false);
				newLayout = await CommandService.instance().execute('toggleNoteList', false);
			}

			newLayout = setLayoutItemProps(newLayout, 'editor', {
				isFullScreen: !isEditorFullScreen,
			});

			// Toggling the sidebar will affect the size of most other on-screen components.
			// Dispatching a window resize event is a bit of a hack, but it ensures that any
			// component that watches for resizes will be accurately notified
			window.dispatchEvent(new Event('resize'));

			context.dispatch({
				type: 'MAIN_LAYOUT_SET',
				value: newLayout,
			});
		},
	};
};
