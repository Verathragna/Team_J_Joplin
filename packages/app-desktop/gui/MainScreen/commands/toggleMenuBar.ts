import { CommandDeclaration, CommandRuntime } from '@joplin/lib/services/CommandService';
import bridge from '../../../services/bridge';
import shim from '@joplin/lib/shim';
import Setting from '@joplin/lib/models/Setting';
import { _ } from '@joplin/lib/locale';

export const declaration: CommandDeclaration = {
	name: 'toggleMenuBar',
	label: () => _('Toggle menu bar'),
};

export const runtime = (): CommandRuntime => {
	return {
		execute: async () => {
			// Defensive code: macOS disallows hiding the menu bar, so ignore this Command
			if (shim.isMac()) return;

			const electronWindow = bridge().window();
			Setting.toggle('hideMenuBar');

			// Now apply the setting
			electronWindow.setAutoHideMenuBar(Setting.value('hideMenuBar'));
			electronWindow.setMenuBarVisibility(!Setting.value('hideMenuBar'));
		},
	};
};
