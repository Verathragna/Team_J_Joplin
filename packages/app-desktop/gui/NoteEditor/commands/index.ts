// AUTO-GENERATED using `gulp buildScriptIndexes`
import * as focusElementNoteBody from './focusElementNoteBody';
import * as focusElementNoteTitle from './focusElementNoteTitle';
import * as focusToolbar from './focusToolbar';
import * as pasteAsText from './pasteAsText';
import * as showLocalSearch from './showLocalSearch';
import * as showRevisions from './showRevisions';

const index: any[] = [
	focusElementNoteBody,
	focusElementNoteTitle,
	focusToolbar,
	pasteAsText,
	showLocalSearch,
	showRevisions,
];

export default index;
// AUTO-GENERATED using `gulp buildScriptIndexes`
