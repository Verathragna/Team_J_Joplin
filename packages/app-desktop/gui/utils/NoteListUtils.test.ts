import NoteListUtils from './NoteListUtils';
import KeymapService from '@joplin/lib/services/KeymapService';
import menuCommandNames from '../menuCommandNames';
import { MenuItem as MenuItemType } from '@joplin/lib/services/commands/MenuUtils';
import initializeCommandService from '../../utils/initializeCommandService';
import { createAppDefaultWindowState } from '../../app.reducer';

jest.mock('../../services/bridge', () => ({
	__esModule: true,
	default: () => ({
		MenuItem: class MenuItem {
			public value: MenuItemType;
			public constructor(value: MenuItemType) {
				this.value = value;
			}
		},
		Menu: class MockMenu {
			public items: MenuItemType[] = [];
			public append(item: MenuItemType) {
				this.items.push(item);
			}
		},
	}),
}));

const mockDispatch = jest.fn();

describe('NoteListUtils', () => {

	beforeEach(() => {
		const mockStore = {
			getState: () => {
				return {
					...createAppDefaultWindowState(),
					settings: {},
				};
			},
		};

		initializeCommandService(mockStore, false);
		const keymapService = KeymapService.instance();
		keymapService.initialize(menuCommandNames());
	});

	it('should show only trash menu options on deleted note', () => {
		const noteIds = ['noteId1'];
		const deletedNote = {
			id: 'noteId1',
			deleted_time: new Date().getTime(),
		};
		const menu = NoteListUtils.makeContextMenu(noteIds, {
			notes: [
				deletedNote,
			],
			dispatch: mockDispatch,
			watchedNoteFiles: [],
			plugins: {},
			inConflictFolder: false,
			customCss: '',
		});

		expect(menu.items).toMatchSnapshot();
	});

	it('should show menu options for normal notes', () => {
		const noteIds = ['noteId1'];
		const normalNote = {
			id: 'noteId1',
		};
		const menu = NoteListUtils.makeContextMenu(noteIds, {
			notes: [
				normalNote,
			],
			dispatch: mockDispatch,
			watchedNoteFiles: [],
			plugins: {},
			inConflictFolder: false,
			customCss: '',
		});

		expect(menu.items).toMatchSnapshot();
	});

	it('should show options when more than one note is selected', () => {
		const noteIds = ['noteId1', 'noteId2'];
		const menu = NoteListUtils.makeContextMenu(noteIds, {
			notes: [
				{ id: 'noteId1' },
				{ id: 'noteId2' },
			],
			dispatch: mockDispatch,
			watchedNoteFiles: [],
			plugins: {},
			inConflictFolder: false,
			customCss: '',
		});

		expect(menu.items).toMatchSnapshot();
	});

	it('should hide all options for encrypted', () => {
		const noteIds = ['noteId1'];
		const encrypted = {
			id: 'noteId1',
			encryption_applied: 1,
		};
		const menu = NoteListUtils.makeContextMenu(noteIds, {
			notes: [
				encrypted,
			],
			dispatch: mockDispatch,
			watchedNoteFiles: [],
			plugins: {},
			inConflictFolder: false,
			customCss: '',
		});

		expect(menu.items).toMatchSnapshot();
	});
});
