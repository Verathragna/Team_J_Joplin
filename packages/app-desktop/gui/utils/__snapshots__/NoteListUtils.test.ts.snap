// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteListUtils should hide all options for encrypted 1`] = `[]`;

exports[`NoteListUtils should show menu options for normal notes 1`] = `
[
  MenuItem {
    "value": {
      "accelerator": "Ctrl+Alt+N",
      "click": [Function],
      "enabled": false,
      "id": "openNoteInNewWindow",
      "label": "Open in new window",
    },
  },
  MenuItem {
    "value": {
      "click": [Function],
      "enabled": false,
      "id": "startExternalEditing",
      "label": "Open in external editor",
    },
  },
  MenuItem {
    "value": {
      "type": "separator",
    },
  },
  MenuItem {
    "value": {
      "accelerator": "Ctrl+Alt+T",
      "click": [Function],
      "enabled": false,
      "id": "setTags",
      "label": "Tags",
    },
  },
  MenuItem {
    "value": {
      "type": "separator",
    },
  },
  MenuItem {
    "value": {
      "click": [Function],
      "enabled": false,
      "id": "toggleNoteType",
      "label": "Switch between note and to-do type",
    },
  },
  MenuItem {
    "value": {
      "click": [Function],
      "enabled": false,
      "id": "moveToFolder",
      "label": "Move to notebook",
    },
  },
  MenuItem {
    "value": {
      "click": [Function],
      "enabled": false,
      "id": "duplicateNote",
      "label": "Duplicate",
    },
  },
  MenuItem {
    "value": {
      "click": [Function],
      "enabled": false,
      "id": "deleteNote",
      "label": "Delete note",
    },
  },
  MenuItem {
    "value": {
      "type": "separator",
    },
  },
  MenuItem {
    "value": {
      "click": [Function],
      "label": "Copy Markdown link",
    },
  },
  MenuItem {
    "value": {
      "click": [Function],
      "label": "Copy external link",
    },
  },
  MenuItem {
    "value": {
      "type": "separator",
    },
  },
  MenuItem {
    "value": {
      "label": "Export",
      "submenu": MockMenu {
        "items": [
          MenuItem {
            "value": {
              "click": [Function],
              "label": "JEX - Joplin Export File",
            },
          },
          MenuItem {
            "value": {
              "click": [Function],
              "label": "RAW - Joplin Export Directory",
            },
          },
          MenuItem {
            "value": {
              "click": [Function],
              "label": "MD - Markdown",
            },
          },
          MenuItem {
            "value": {
              "click": [Function],
              "label": "MD - Markdown + Front Matter",
            },
          },
          MenuItem {
            "value": {
              "click": [Function],
              "label": "HTML - HTML File",
            },
          },
          MenuItem {
            "value": {
              "click": [Function],
              "label": "HTML - HTML Directory",
            },
          },
          MenuItem {
            "value": {
              "accelerator": null,
              "click": [Function],
              "enabled": false,
              "id": "exportPdf",
              "label": "PDF - PDF File",
            },
          },
        ],
      },
    },
  },
]
`;

exports[`NoteListUtils should show only trash menu options on deleted note 1`] = `
[
  MenuItem {
    "value": {
      "click": [Function],
      "enabled": false,
      "id": "restoreNote",
      "label": "Restore note",
    },
  },
  MenuItem {
    "value": {
      "accelerator": null,
      "click": [Function],
      "enabled": false,
      "id": "permanentlyDeleteNote",
      "label": "Permanently delete note",
    },
  },
]
`;

exports[`NoteListUtils should show options when more than one note is selected 1`] = `
[
  MenuItem {
    "value": {
      "accelerator": "Ctrl+Alt+T",
      "click": [Function],
      "enabled": false,
      "id": "setTags",
      "label": "Tags",
    },
  },
  MenuItem {
    "value": {
      "type": "separator",
    },
  },
  MenuItem {
    "value": {
      "click": [Function],
      "label": "Switch to note type",
    },
  },
  MenuItem {
    "value": {
      "click": [Function],
      "label": "Switch to to-do type",
    },
  },
  MenuItem {
    "value": {
      "click": [Function],
      "enabled": false,
      "id": "moveToFolder",
      "label": "Move to notebook",
    },
  },
  MenuItem {
    "value": {
      "click": [Function],
      "enabled": false,
      "id": "duplicateNote",
      "label": "Duplicate",
    },
  },
  MenuItem {
    "value": {
      "click": [Function],
      "enabled": false,
      "id": "deleteNote",
      "label": "Delete note",
    },
  },
  MenuItem {
    "value": {
      "type": "separator",
    },
  },
  MenuItem {
    "value": {
      "click": [Function],
      "label": "Copy Markdown link",
    },
  },
  MenuItem {
    "value": {
      "type": "separator",
    },
  },
  MenuItem {
    "value": {
      "label": "Export",
      "submenu": MockMenu {
        "items": [
          MenuItem {
            "value": {
              "click": [Function],
              "label": "JEX - Joplin Export File",
            },
          },
          MenuItem {
            "value": {
              "click": [Function],
              "label": "RAW - Joplin Export Directory",
            },
          },
          MenuItem {
            "value": {
              "click": [Function],
              "label": "MD - Markdown",
            },
          },
          MenuItem {
            "value": {
              "click": [Function],
              "label": "MD - Markdown + Front Matter",
            },
          },
          MenuItem {
            "value": {
              "click": [Function],
              "label": "HTML - HTML Directory",
            },
          },
          MenuItem {
            "value": {
              "accelerator": null,
              "click": [Function],
              "enabled": false,
              "id": "exportPdf",
              "label": "PDF - PDF File",
            },
          },
        ],
      },
    },
  },
]
`;
